spec/models/user_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+describe User do
+  it { should have_valid(:first_name).when("John", "Sally") }
+  it { should_not have_valid(:first_name).when(nil, "") }
+
+  it { should have_valid(:last_name).when("Smith", "Kelly") }
+  it { should_not have_valid(:last_name).when(nil, "") }
+
+  it { should have_valid(:email).when('user@example.com',
```

Comment:
Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.